extensions/api/data-management/contractnegotiation/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractnegotiation/model/ContractNegotiationDto.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *   Florian Rusch - Initial API and Implementation
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.contractnegotiation.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.ContractNegotiation.Type;
+
+@JsonDeserialize(builder = ContractNegotiationDto.Builder.class)
+public class ContractNegotiationDto {
+    private String id;
+    private String counterPartyAddress;
+    private String protocol;
+    private final Type type = Type.CONSUMER;
+    private String state;
+    private String errorDetail;
+    private String contractAgreementId; // is null until state == CONFIRMED
+
+    private ContractNegotiationDto() {
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getCounterPartyAddress() {
+        return counterPartyAddress;
+    }
+
+    public String getProtocol() {
+        return protocol;
+    }
+
+    public Type getType() {
+        return type;
+    }
+
+    public String getState() {
+        return state;
+    }
+
+    public String getErrorDetail() {
+        return errorDetail;
+    }
+
+    public String getContractAgreementId() {
+        return contractAgreementId;
+    }
+
+    @JsonPOJOBuilder(withPrefix = "")
+    public static final class Builder {
+        private final ContractNegotiationDto dto;
+
+        private Builder() {
+            dto = new ContractNegotiationDto();
+        }
+
+        @JsonCreator
+        public static Builder newInstance() {
+            return new Builder();
+        }
+
+        public Builder id(String id) {
+            dto.id = id;
+            return this;
+        }
+
+        public Builder counterPartyAddress(String counterPartyAddress) {
+            dto.counterPartyAddress = counterPartyAddress;
+            return this;
+        }
+
+        public Builder protocol(String protocol) {
+            dto.protocol = protocol;
+            return this;
+        }
+
+        public Builder state(String state) {
+            dto.state = state;
+            return this;
+        }
+
+        public Builder errorDetail(String errorDetail) {
+            dto.errorDetail = errorDetail;
+            return this;
+        }
+
+        public Builder contractAgreementId(String contractAgreementId) {
+            dto.contractAgreementId = contractAgreementId;
+            return this;
+        }
+
+        public Builder type(Type type) {
+            if (type != Type.CONSUMER) {
+                throw new IllegalArgumentException("The type can only be CONSUMER");
```

Comment:
DTOs should not validate input, especially not in builder methods.

Reply:
I'm totally with you! But how should I then handle the type attribute? My first attempt was to not implement the type method in the Builder class because it should be set automatically at new object anyway. But then I got this error from the serialization test:
```
Unrecognized field "type" (class org.eclipse.dataspaceconnector.api.datamanagement.contractnegotiation.model.ContractNegotiationDto$Builder), not marked as ignorable (6 known properties: "state", "contractAgreementId", "protocol", "errorDetail", "counterPartyAddress", "id"])
 at [Source: (String)"{"id":"test-id","counterPartyAddress":"test-counter-party-address","protocol":"test-protocol","type":"CONSUMER","state":"test-state","errorDetail":"test-error-detail","contractAgreementId":"test-contract-agreement-id"}"; line: 1, column: 103] (through reference chain: org.eclipse.dataspaceconnector.api.datamanagement.contractnegotiation.model.ContractNegotiationDto$Builder["type"])
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "type" (class org.eclipse.dataspaceconnector.api.datamanagement.contractnegotiation.model.ContractNegotiationDto$Builder), not marked as ignorable (6 known properties: "state", "contractAgreementId", "protocol", "errorDetail", "counterPartyAddress", "id"])
 at [Source: (String)"{"id":"test-id","counterPartyAddress":"test-counter-party-address","protocol":"test-protocol","type":"CONSUMER","state":"test-state","errorDetail":"test-error-detail","contractAgreementId":"test-contract-agreement-id"}"; line: 1, column: 103] (through reference chain: org.eclipse.dataspaceconnector.api.datamanagement.contractnegotiation.model.ContractNegotiationDto$Builder["type"])
```

The problem is, that he takes all fields from the json and tries to call methods with the same name on the Builder object -> Kabäm there is no function for the type.

A idea that comes up in my mind while writing this: Should I simply implement this?

```java
        public Builder type(Type type) {
            return this;
        }
```

Reply:
there _should_ be a method to set the type. the type is _not_ fixed, it simply is initialized with `CONSUMER` by default. 

As a bit of general information: the `@JsonIgnore` annotation is used to exclude fields from serialization/deserialization, but that's _not_ the case here.

Reply:
Ahhh okay - interpreted it as been a fix value. Will fix it